src/ffishjs.cpp

```diff
@@ -263,6 +263,34 @@ class Board {
     return pos.game_ply();
   }
 
+  int game_result() const {
+    Value result;
+    bool gameEnd;
+    if (MoveList<LEGAL>(pos).size() > 0)
+      return VALUE_NONE;
+    gameEnd = pos.is_immediate_game_end(result);
+    if (!gameEnd)
+        result = pos.checkers() ? pos.checkmate_value() : pos.stalemate_value();
+    return result;
+  }
+
+  bool is_immediate_game_end() const {
+    return pos.is_immediate_game_end();
+  }
+
+  bool is_optional_game_end() const {
+    return pos.is_optional_game_end();
+  }
+
+  // Need to use different name than has_insufficient_material
+  bool has_insufficient_material_ffish(bool turn) const {
```

Comment:
Please remove comment  
```C++
// Need to use different name than has_insufficient_material
```
 and remove suffix `_ffish`.
 
Below you can use
 ```C++
 return Stockfish::has_insufficient_material(turn ? WHITE : BLACK, pos);
 ```
 to avoid ambiguity.

Reply:
Also please rename into `is_insufficient_material()`.

The **ffish.js** library aims to have a syntax similar to python-chess in cases where this is possible.
In python-chess this function is called

```
is_insufficient_material() → bool[source]
    Checks if neither side has sufficient winning material (has_insufficient_material()).
```    
* https://python-chess.readthedocs.io/en/latest/core.html#outcome

If you like, you can also add
```
has_insufficient_material(color: chess.Color) → bool
```
accordingly.